common/lib/xmodule/xmodule/js/src/video/01_initialize.js

```diff
@@ -169,6 +169,10 @@ function(VideoPlayer, i18n, moment, _) {
                     _oldOnYouTubeIframeAPIReady = window.onYouTubeIframeAPIReady || undefined;
 
                     window.onYouTubeIframeAPIReady = function() {
+                        if (!window.onYouTubeIframeAPIReady.resolve) {
+                            window.onYouTubeIframeAPIReady.resolve = _youtubeApiDeferred.resolve;
+                        }
+
                         window.onYouTubeIframeAPIReady.resolve();
                     };
```

Comment:
OK so I think what's happening here is that Google Tag Manager overwrites `window.onYouTubeIframeAPIReady` in order to provide tracking of YouTube, and it does something like this:

```
const oldReady = window.onYouTubeIframeAPIReady;
window.onYouTubeIframeAPIReady = function() {
    ... GTM code...
    oldReady();
}
```

This should normally work fine, but because of how the function above is written, it expects to find the `.resolve` property on the `window.onYouTubeIframeAPIReady` function, which has now changed.

I see why your fix works, but I think it's potentially confusing as at first glance it seems to be redundant. (Why would `window.onYouTubeIframeAPIReady.resolve` be undefined when it's set on the very next line after this function is declared?)

I wonder if we can take a much simpler approach? I think that this should work both with or without GTM, and I think it will be less confusing for anyone who has to work with this code in the future.

```javascript
setupOnYouTubeIframeAPIReady = function() {
    const _oldOnYouTubeIframeAPIReady = window.onYouTubeIframeAPIReady || undefined;

    window.onYouTubeIframeAPIReady = () => {
        // Call all the 'done()' handlers attached to _youtubeApiDeferred:
        _youtubeApiDeferred.resolve();
        if (_oldOnYouTubeIframeAPIReady) {
            _oldOnYouTubeIframeAPIReady();
        }
    };
};
```

What do you think?

Reply:
You are definitely right. Your advised solution will be more predictable and transparent.
But when I tried to review your solution, I couldn't reproduce this bug on the maple release again. I suppose that it might be fixed on the edX or google (GTM developer team) side during my PR was opened and was waited for the review.

Reply:
In that case, feel free to close this if you think it's no longer needed. I'd also be happy to merge it now if you're interested in the simpler approach that I suggested, if you can confirm that at least it doesn't introduce any regressions. It's possible that the bug is not really fixed, but whether or not it happens depends on how the various scripts load and in what order, which may make it hard to reproduce consistently.

Reply:
We checked it and this bug was fixed for the latest release. Based on that, I'll close this PR.
Thanks a lot for the communications and your suggestions.

Reply:
Thank you for taking the time to submit a fix for the issue, @ihor-romaniuk! I'm glad it's fixed one way or another.